tools/state-viewer/src/state_dump.rs

```diff
@@ -348,12 +373,94 @@ mod test {
             last_block.header().clone(),
             &near_config,
             Some(&records_file.path().to_path_buf()),
+            None,
         );
         let new_genesis = new_near_config.genesis;
         assert_eq!(new_genesis.config.validators.len(), 2);
         validate_genesis(&new_genesis);
     }
 
+    /// Test that we respect the specified account ID list in dump_state.
+    #[test]
+    fn test_dump_state_respect_select_account_ids() {
+        let epoch_length = 4;
```

Comment:
these 40 lines of code are (I think) the same in all the test cases -- could you refactor them into a separate method instead ? (can be done in separate PR)

Reply:
There is still some difference. In my case I set up more accounts than the rest of the tests.